src/controller/src/lib.rs

```diff
@@ -267,279 +119,20 @@ enum Readiness {
 
 /// A client that maintains soft state and validates commands, in addition to forwarding them.
 pub struct Controller<T = mz_repr::Timestamp> {
-    build_info: &'static BuildInfo,
-    storage_controller: Box<dyn StorageController<Timestamp = T>>,
-    compute_orchestrator: Arc<dyn NamespacedOrchestrator>,
-    computed_image: String,
-    compute: BTreeMap<ComputeInstanceId, ComputeController<T>>,
+    pub storage: Box<dyn StorageController<Timestamp = T>>,
+    pub compute: ComputeController<T>,
```

Comment:
Not convinced that we want to have field access here or a function providing access. I'd feel better about it if the `Controller` lived in `coord`, especially because all uses are from `mz-adapter` or `environmentd` (`new`).

Reply:
I don't have a strong opinion here, happy to revert to accessor methods again.

Can you explain what you don't like about providing direct field access? AFAICT the only functional difference is that clients can move the storage and compute controller out of the global controller, which doesn't seem useful but also not much of an issue.

Reply:
Happy to leave it as-is! We can think about inlining the `Controller` with coord later, but it doesn't need to happen as part of this PR.